provider/cpu_utils.py

```diff
@@ -98,8 +100,18 @@ def check_if_vm_vcpu_topology_match(session, os_type, cpuinfo):
                               " returned by guest:\n%s", out)
                 return False
 
+    threads_matched = cpuinfo.threads*cpuinfo.cores == threads
```

Comment:
`threads_matched = threads//cores == cpuinfo.threads`?
As we know "threads per socket" and "cores per socket" inside guest, so I think this formula is more logical?

Reply:
I agree, it is more human readable, let's change that.

Reply:
Ah sorry, I just realized: it won't work. Let me revert a bit. 

We have two problems:
1) Incorrect behaviour for Windows guests for QEMU < 8.1 => so we agreed disable check for threads
2) Inconsistent behaviour between Win and Linux guests for QEMU >= 8.1 => we still need to address this

Reply:
Yes @lkotek, but as @yanan-fu mentioned, you will move the code into windows part, so I think the change may not affect Linux?

Reply:
Soory @PaulYuuu, I was mixing multiple things together. 

The change you proposed here `threads_matched = threads//cores == cpuinfo.threads` is just fine.

The tests with `RHEL` and `Win2022` guests passed in case of `qemu-8.2`:

```
JOB LOG    : /root/avocado/job-results/job-2024-01-05T05.56-f95fd67/job.log
 (1/2) Host_RHEL.m9.u4.v0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.cpu_device_hotpluggable.hotplug.single_vcpu.with_reboot.shell_reboot.q35: STARTED
 (1/2) Host_RHEL.m9.u4.v0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.cpu_device_hotpluggable.hotplug.single_vcpu.with_reboot.shell_reboot.q35: PASS (170.95 s)
 (2/2) Host_RHEL.m9.u4.v0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.cpu_device_hotpluggable.hotplug.single_vcpu.with_reboot.shell_reboot.q35: STARTED
 (2/2) Host_RHEL.m9.u4.v0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.cpu_device_hotpluggable.hotplug.single_vcpu.with_reboot.shell_reboot.q35: PASS (299.11 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```